### DISCUSSION TITLE
TreeSelect 是否可以和 Select 控件一样添加一个 labelRender 属性

### DISCUSSION THREAD

**Discussion by ekuku1984** [ant-design/ant-design#51764] [Ideas]
今天有个需求， 就 TreeSelect 控件选择完数据后， 需要把整个选定数据的上级全部显示出来。

经过查看文档，  TreeSelect 没有像 Select 控件一样的 labelRender, 有一个 treeTitleRender, 经过测试， 使用 treeTitleRender 会修改下拉树上显示的节点， 这个方法不适合来实现这个需求。

还看到一个 labelInValue 属性， 把这个设置为 true, 并在每一项中设置一个 label 好像也效果。

还是希望添加一个 labelRender 属性， 除了显示全路径外， 还有可能会给选定的值添加一个链接， 用来显示选定数据的详细信息.

**Comment by petercat-assistant** [comment]
目前，`TreeSelect` 组件确实没有像 `Select` 组件那样的 `labelRender` 属性。你提到的 `treeTitleRender` 是用于自定义下拉树节点的显示，而不是用于自定义选中项的显示。

对于你提到的需求，使用 `labelInValue` 属性并在每项中设置一个 `label` 是一种可行的解决方案，但如果你希望有更灵活的自定义显示功能，比如显示全路径或添加链接，目前可能需要通过其他方式实现。

关于添加 `labelRender` 属性的建议，我会建议你在 GitHub 上的 ant-design 仓库中查看是否有类似的功能请求，或者提出一个新的功能请求，以便开发团队可以考虑在未来的版本中加入这个功能。

如果你需要进一步的帮助或有其他问题，请在评论中描述你的问题并 @petercat-assistant 以开始与我对话。

**Comment by Wxh16144** [comment]
https://github.com/ant-design/ant-design/issues?q=is%3Aissue+TreeSelect+labelRender

**Reply by ekuku1984** [reply]
谢谢回复！！， 这种方式确实可以解决目前的问题， 但是还是希望有 labelRender 这个属性， 不然在处理数据的时候， 我需要提供把所有节点的全路径全部计算一遍。

通过搜索相关 treeselect 相关的信息， 大家对 labelRender 这个属性应该还是有很大的需求的
